### PR TITLE
Fix the sidecar mutate when using multiple PVCs in a single pod

### PR DESCRIPTION
**Pull request by zxh326**

### CODE REVIEW

File: pkg/juicefs/mount/builder/vci-serverless.go [hunk 1]
```diff
@@ -64,13 +71,27 @@ func NewVCIBuilder(setting *config.JfsSetting, capacity int64, app corev1.Pod, p
...
+	propagations = append(propagations, VCIPropagationStruct{
+		Container: mountContainerName,
+		MountPath: r.jfsSetting.MountPath,
+	})
+	if v, ok := r.app.Annotations[VCIPropagation]; ok {
+		var vciPropagations []VCIPropagationStruct
+		if err := json.Unmarshal([]byte(v), &vciPropagations); err == nil {
+			propagations = append(propagations, vciPropagations...)
+		} else {
+			klog.Errorf("failed to unmarshal VCIPropagation annotation: %v", err)
```
Comment (timfeirg) [inline pkg/juicefs/mount/builder/vci-serverless.go:90]: any change we print the original annotations string in this log?

```suggestion
			klog.Errorf("failed to unmarshal VCIPropagation annotation %s %v", v, err)
```

File: pkg/juicefs/mount/builder/vci-serverless.go [hunk 2]
```diff
@@ -64,13 +71,27 @@ func NewVCIBuilder(setting *config.JfsSetting, capacity int64, app corev1.Pod, p
 func (r *VCIBuilder) NewMountSidecar() *corev1.Pod {
```
Comment (timfeirg) [inline pkg/juicefs/mount/builder/vci-serverless.go:71]: if not too much trouble, could you apply the changes to all `pkg/juicefs/mount/builder/*serverless.go` codes? I imagine the code will be all similar?

this way, if tests pass for `pkg/juicefs/mount/builder/serverless.go`, we're quite confident that it's going to work for other platforms as well?

Comment (zxh326) [reply pkg/juicefs/mount/builder/vci-serverless.go:71]: only vci requires this annotation

File: pkg/juicefs/mount/builder/vci-serverless.go (hunk 1 above)
Comment (zxh326) [reply pkg/juicefs/mount/builder/vci-serverless.go:90]: emmm...I believe there is no need to print

File: pkg/juicefs/mount/builder/vci-serverless.go (hunk 2 above)
Comment (zxh326) [reply pkg/juicefs/mount/builder/vci-serverless.go:71]: *serverless.go have a lot of similar code, which can be optimized later.

**Review by timfeirg — APPROVED** [review]
测通以后自行合并

**Pull request merged**